x/ccv/consumer/keeper/relay.go

```diff
@@ -213,6 +216,16 @@ func (k Keeper) SendPackets(ctx sdk.Context) {
 			k.Logger(ctx).Error("cannot send IBC packet; leaving packet data stored:", "type", p.Type.String(), "err", err.Error())
 			break
 		}
+		// If the packet that was just sent was a Slash packet, set the waiting on slash reply flag.
+		// This flag will be toggled false again when consumer hears back from provider. See OnAcknowledgementPacket below.
+		if p.Type == ccv.SlashPacket {
+			k.UpdateSlashRecordOnSend(ctx)
+			// Break so slash stays at head of queue
```

Comment:
Is the idea that we we want to `break` here, so that potential `VSCMature` packets that come after the `SlaskPacket` do **not** take precedence?
 It might be useful to add in the comment the exact reason we want the `SlashPacket` in the head of the queue.

Reply:
The idea is to follow [this design](https://cosmos.github.io/interchain-security/adrs/adr-008-throttle-retries#consumer-changes), we're breaking so that the sending of any other packet is blocked until the slash packet is handled

Hopefully [this change ](https://github.com/cosmos/interchain-security/commit/9d111a87c68ac85046d84b7364ac30b711cec144) can help clear things up